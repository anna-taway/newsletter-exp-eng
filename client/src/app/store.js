import { configureStore } from "@reduxjs/toolkit";
import registrationReducer from "../features/userRegistration/userRegistrationSlice";
import productsReducer from '../features/products/productsSlice';


const store = configureStore({
	reducer: {
		registration: registrationReducer,
		products: productsReducer
	},
});

export default store;