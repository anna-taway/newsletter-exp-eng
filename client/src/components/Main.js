import React from 'react';
import { Route, Switch } from 'react-router-dom';
import ProductsListView from './ProductsListView';
import UserRegistrationView from './UserRegistrationView';

const Main = () => {
  return (
    <Switch>
        <Route path="/" exact component={ProductsListView}></Route>
        <Route path="/user" exact component={UserRegistrationView}></Route>
    </Switch>
  )
}

export default Main;


