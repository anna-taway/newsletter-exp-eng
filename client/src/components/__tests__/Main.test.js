import { render, screen } from '@testing-library/react';
import { create } from "react-test-renderer";
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';

import '@testing-library/jest-dom/extend-expect';

import { Provider } from 'react-redux';
import store from '../../app/store';
import Main from '../Main';

describe('Main app rendering/navigating', () => {
  test('Main app rendering/navigating', () => {
    const history = createMemoryHistory()
    render(
      <Provider store={store}>
          <Router history={history}>
              <Main />
          </Router>
      </Provider>
      
    )
    // verify page content for expected route
    expect(screen.getByText(/Some news about EE and QBE/i)).toBeInTheDocument()
  
    const leftClick = { button: 0 }
    userEvent.click(screen.getByText(/Subscribe/i), leftClick)
  
    // check that the content changed to the new page
    expect(screen.getByLabelText(/Username/i)).toBeInTheDocument()
  });

  test("Main component matches the snapshot", () => {
    const history = createMemoryHistory()
    const main = create(
      <Provider store={store}>
        <Router history={history}>
          <Main />
        </Router>
      </Provider>
    );
    expect(main.toJSON()).toMatchSnapshot();
  });
})

