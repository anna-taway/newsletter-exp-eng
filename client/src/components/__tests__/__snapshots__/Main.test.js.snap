// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main app rendering/navigating Main component matches the snapshot 1`] = `
Array [
  <div
    className="makeStyles-productsContent-23"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <h1
        className="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Newsletter
      </h1>
      <p
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
      >
        Some news about EE and QBE
      </p>
      <div
        className="makeStyles-productsButtons-24"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <a
                  color="inherit"
                  href="/user"
                  onClick={[Function]}
                >
                  Subscribe
                </a>
              </span>
            </button>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <a
                  color="inherit"
                  href="/"
                  onClick={[Function]}
                >
                  Find more
                </a>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiContainer-root makeStyles-cardGrid-25 MuiContainer-maxWidthMd"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    />
  </div>,
]
`;
